src/hotspot/os/windows/os_windows.cpp

```diff
@@ -2909,22 +2909,25 @@ LONG WINAPI topLevelVectoredExceptionFilter(struct _EXCEPTION_POINTERS* exceptio
 
 #if defined(USE_VECTORED_EXCEPTION_HANDLING)
 LONG WINAPI topLevelUnhandledExceptionFilter(struct _EXCEPTION_POINTERS* exceptionInfo) {
-  if (InterceptOSException) goto exit;
-  DWORD exception_code = exceptionInfo->ExceptionRecord->ExceptionCode;
+  if (InterceptOSException) {
+    goto result;
+  } else {
+    DWORD exceptionCode = exceptionInfo->ExceptionRecord->ExceptionCode;
```

Comment:
No you brought the goto back! Why? goto's are bad.

Reply:
Sorry, was experimenting different ways of doing it. I guess the duplicated return statement is the way to go, as Thomas suggested